libs/traits/src/lib.rs

```diff
@@ -804,6 +857,120 @@ pub mod ops {
 
 	impl<T: CheckedAdd> EnsureAdd for T {}
 	impl<T: CheckedSub> EnsureSub for T {}
-	impl<T: CheckedMul> EnsureMul for T {}
+	impl<T: CheckedMul + Signum> EnsureMul for T {}
 	impl<T: CheckedDiv> EnsureDiv for T {}
+
+	/// Performs self addition that returns `ArithmeticError::Overflow` instead of
+	/// wrapping around on overflow.
+	pub trait EnsureAddAssign: EnsureAdd {
+		/// Adds two numbers overwriting the left hand one, checking for overflow.
+		/// If overflow happens, `ArithmeticError::Overflow` is returned.
+		///
+		/// ```
+		/// use cfg_traits::ops::EnsureAddAssign;
+		/// use sp_runtime::{DispatchResult, ArithmeticError, DispatchError};
+		///
+		/// fn extrinsic() -> DispatchResult {
+		///     let mut max = u32::MAX;
+		///     max.ensure_add_assign(&1)?;
+		///     Ok(())
+		/// }
+		///
+		/// assert_eq!(extrinsic(), Err(DispatchError::Arithmetic(ArithmeticError::Overflow)));
+		/// ```
+		fn ensure_add_assign(&mut self, v: &Self) -> Result<(), ArithmeticError> {
```

Comment:
Given that this is generic and could be used with a signed number type, couldn't this operation also lead to an Underflow error case? For example if the left hand side is the MIN of said type and right hand side is a some non-zero negative number?

Reply:
Same question applies to the `ensure_sub_assign`

Reply:
Totally right. I'm going to update it.